grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy

```diff
@@ -173,20 +177,20 @@ class ReportController {
                     expirationDate: it.inventoryItem.expirationDate,
                     binLocation   : it?.binLocation?.name ?: "Default Bin",
                     quantity      : formatNumber(number: it.quantity),
-                    unitCost      : formatNumber(number: it.unitCost),
-                    totalValue     : formatNumber(number: it.totalValue)
+                    unitCost      : hasRoleFinance ? formatNumber(number: unitCost) : null,
+                    totalValue    : hasRoleFinance ? formatNumber(number: it.quantity * unitCost) : null
```

Comment:
I would recommend keeping the logic hidden behind these convenience methods. 
```
unitCost      : hasRoleFinance ? formatNumber(number: it.unitCost) : null,
totalValue    : hasRoleFinance ? formatNumber(number: it.totalValue) : null
```

Reply:
Or justify the switch to moving the logic here.

Reply:
good point